### ISSUE TITLE
Revamp "key" mode in mg_show() using MgObject dictionaries

### ISSUE THREAD

**Issue by balintlaczko** [fourMs/MGT-python#215]
Because it introduces ambiguity when the image/video render functions use overwrite protection. Previously we always overwrote every result, so it was trivial to guess what filename to look for by only providing a "key". For example, `key='average'` would look for _myvideo_average.png_. However with the overwrite protection enabled, if we run the average image function repeatedly, we will get multiple files, like:
- _myvideo_average.png_
- _myvideo_average_0.png_
- _myvideo_average_1.png_
- etc...

And in that case it is impossible to guess which file is the one to show:
```python
mg = musicalgestures.MgObject('myvideo.avi')
mg.average()
mg.motion()
mg.show(key='average') #???
mg.show(key='motion') #???
```

The much better practice is to save the rendered results to variables, and call `show()` on them:
```python
mg = musicalgestures.MgObject('myvideo.avi')
average = mg.average()
motion = mg.motion()
average.show()
motion.show()
```

In the latter case there is no ambiguity whatsoever about the correct file to show, since the functions always return them.

Actually if we consider that `average` returns an `MgImage` object, and `motion` returns an `MgObject`, plus that `show` returns the `self` it is called on (and that `show` runs now in a non-blocking, asynchronous way), we can further simplify the above code:
```python
mg = musicalgestures.MgObject('myvideo.avi')
average = mg.average().show()
motion = mg.motion().show()
```

The "key" mode should also get removed from the intro notebook (and the documentation).

**Comment by balintlaczko** [comment]
Okay, not so fast. If MgObject dictionaries get implemented (issue #216) we can actually keep (revamp) the key-mode.

**Comment by balintlaczko** [comment]
This is now implemented. On the surface it looks the same, the interaction is the same as before (using `key`s or just calling it on an object), but under the hood it now queries the parent MgObject, instead of relying on file naming conventions.
